status.w: rework to support Darwin/BSD

jasperla opened the pull request:
### What does this PR do?

This adds support for Darwin/FreeBSD/OpenBSD where the output
does not contain several fields that Linux emits.

While here it adds the `from` field which is reliably populated in the BSD implementation. Local logins are tracked as `-` or  the X Windows display.

Due to the fact the Linux implementation of `w` emits nothing for local logins it makes the output harder to parse. So skip printing `from` on Linux entirely.

### Previous Behavior

```
salt.localdomain:
    |_
      ----------
      idle:
          1:08PM
      jcpu:
          0
      login:
          123.123.123.123
      pcpu:
          tmux:
      tty:
          p1
      user:
          jasper
      what:
          client (/tmp/tmux-0/default)
```

It would end up consuming the `what` column to fill out non-existent fields (`jcpu` and `pcpu`).

### New Behavior

```
salt.localdomain:
    |_
      ----------
      from:
          123.123.123.123
      idle:
          0
      login:
          1:08PM
      tty:
          p1
      user:
          jasper
      what:
          tmux: client (/tmp/tmux-0/default)
```

### Tests written?

No.

### Commits signed with GPG?

Yes

rallytime commented:
hi @jasperla - We just completed the feature freeze for the `oxygen` branch. Anything that is a new feature or a rewriting existing code should be added to the `develop` branch for inclusion in the `Fluorine` feature release. Can you move this to the `develop` branch?

I know we've been moving several of your recent PRs from `develop` to `oxygen` lately. This was because those changes were submitted before the feature freeze deadline and we wanted to make sure they were included in `oxygen`. Now that were starting to test and harden the `oxygen` branch, we can no longer accept feature additions.

This also needs some tests as well, once you have a chance to move this over to `develop`.

Thank you so much for all of your great work recently!

jasperla committed:
status.w: rework to support Darwin/BSD

This adds support for Darwin/FreeBSD/OpenBSD where the output
does not contain several fields that Linux emits.

While here add the `from` field on BSD which is reliably populated
in the BSD implementation. Local logins are tracked as `-` or
the X Windows display.
For Linux skip `from` completely as it may be empty on local logins.

jasperla commented:
I've moved this over to `develop` and added the missing tests too. Thanks for your feedback and merges!

rallytime commented:
Great! Thank you @jasperla!
